test/response.test.js

```diff
@@ -514,7 +514,10 @@ test('should fail to set header due to missing formatter', function(t) {
     CLIENT.get(join(LOCALHOST, '/11'), function(err, _, res) {
         t.ifError(err);
         t.equal(res.statusCode, 200);
-        t.equal(res.headers['content-type'], 'application/octet-stream');
+        t.equal(
+            res.headers['content-type'],
+            'application/octet-stream; charset=utf-8'
+        );
```

Comment:
Binary content types should not include a charset as it only makes sense for textual content types.

Reply:
That is true. Didn't think of that. Having a list of different mime-types that should default to utf-8 is not convenient either keeping in mind all combinations. How about a blacklist of mime-types where not to format with utf-8?

Reply:
The list of binary types is also unbounded. I think it makes most sense to make the formatter to be responsible for picking a default / forcing a particular charset.

Reply:
The Python WebOb library uses an include list for setting a default charset:

>     * The ``default_charset`` is used as the default character set to return on
>       the ``Content-Type`` header, if the ``Content-Type`` allows for a
>       ``charset`` parameter. Currently the only ``Content-Type``'s that allow
>       for a ``charset`` are defined to be: ``text/*``, ``application/xml``, and
>       ``*/*+xml``. Any other ``Content-Type``'s will not have a ``charset``
>       added.

https://github.com/Pylons/webob/blob/master/src/webob/response.py#L142

We'd probably need to add at least "application/javascript", "application/json", and "*/*+json" to that list.

Maintaining such a list does seem very inconvenient. Attaching the default to the formatter somehow seems more extensible.

Reply:
Would this help? https://www.npmjs.com/package/mime-types

```js
mime.charset('text/markdown') // 'UTF-8'
```